configuration/configuration.go

```diff
@@ -11,7 +11,10 @@ import (
 )
 
 // Configuration is a versioned registry configuration, intended to be provided by a yaml file, and
-// optionally modified by environment variables
+// optionally modified by environment variables.
+//
+// Note that yaml field names should never include _ characters, since this is the separator used
```

Comment:
Added a check for this `overwriteStruct` with a panic. It will cause unit tests to fail if someone makes this mistake.